opal/util/argv.c

```diff
@@ -343,7 +343,7 @@ char *opal_argv_join_range(char **argv, size_t start, size_t end, int delimiter)
 
     /* Bozo case */
 
-    if (NULL == argv || NULL == argv[0] || (int)start > opal_argv_count(argv)) {
+    if (NULL == argv || NULL == argv[0] || (int)start >= opal_argv_count(argv)) {
         return strdup("");
```

Comment:
We need to be consistent in this function when something didn't go as planned: either we return `strdup("")` or NULL, but consistently.

Reply:
I think the point of returning NULL is the assumption that if `malloc` returns NULL we're out-of-memory, so `strdup` won't work either.